Lib/http/server.py

```diff
@@ -1251,6 +1252,46 @@ def run_cgi(self):
                 self.log_message("CGI script exited OK")
 
 
+class HTTPSServer(HTTPServer):
+    def __init__(self, server_address, RequestHandlerClass,
+                 bind_and_activate=True, *, certfile, keyfile=None,
+                 password=None, alpn_protocols=None):
+        try:
+            import ssl
+        except ImportError:
+            raise RuntimeError("SSL module is missing; HTTPS support is unavailable")
+
+        self.ssl = ssl
+        self.certfile = certfile
+        self.keyfile = keyfile
+        self.password = password
+        # Support by default HTTP/1.1
+        self.alpn_protocols = alpn_protocols or ["http/1.1"]
```

Comment:
```suggestion
        self.alpn_protocols = ["http/1.1"] if alpn_protocols is None else alpn_protocols
```

Someone might want `alpn_protocols=[]` for tests.